x/ecocredit/client/core/proposal.go

```diff
@@ -0,0 +1,78 @@
+package core
+
+import (
+	"fmt"
+	"strconv"
+
+	govclient "github.com/cosmos/cosmos-sdk/x/gov/client"
+	"github.com/cosmos/cosmos-sdk/x/gov/client/rest"
+	"github.com/spf13/cobra"
+
+	"github.com/cosmos/cosmos-sdk/client"
+	"github.com/cosmos/cosmos-sdk/client/tx"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/x/gov/client/cli"
+	"github.com/cosmos/cosmos-sdk/x/gov/types"
+
+	"github.com/regen-network/regen-ledger/x/ecocredit/core"
+)
+
+var CreditTypeProposalHandler = govclient.NewProposalHandler(TxCreditTypeProposalCmd, func(context client.Context) rest.ProposalRESTHandler {
+	return rest.ProposalRESTHandler{ // TODO: what to do here??
+		SubRoute: "",
+		Handler:  nil,
+	}
+})
+
+func TxCreditTypeProposalCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "credit-type-proposal [proposal-title] [proposal-description] [credit_type_abbreviation] [credit_type_name] [units] [precision] [flags]",
```

Comment:
This seems like a lot of positional arguments. I think we shouldn't even implement this and just let people submit a JSON proposal. I don't expect this to be called more than 5-10 times ever in production

Reply:
is that how most folks submit proposals? i could change it to a single argument (json file)

Reply:
I think aaron is referring to the gov command:

```
regen tx gov submit-proposal --proposal="path/to/proposal.json" --from mykey
```

Where `proposal.json` includes:

```
{
  "title": "Add Biochar Credit Type",
  "description": "Add Biochar credit type to the list of allowed credit types",
  "type": "CreditTypeProposal",
  "credit_type": {
    "abbreviation": "BC",
    "name": "biochar",
    "unit": "tonnes of organic carbon",
    "precision": 6,
  },
  "deposit": "10regen"
}
```

I'm not sure how `type` would work with this custom implementation though. If this is not possible, I do think we should limit the number of arguments and use a json file.

Reply:
I'm not sure if the above works and it should be tested before removing the command implemented here. If not, I think a json or yaml file might be nice.

Reply:
ah ok didn't know of that one. removed the cli command - https://github.com/regen-network/regen-ledger/pull/1015/commits/fa42e6d04ed4d1482337cb9516ad28a50cd7948e

Reply:
Does it work?

Reply:
didnt see your last comment 🥲, wasn't able to get this working. maybe we can take a look tomorrows standup? can just revert the removal if we cant get it going

Reply:
Sounds good. Let's try tomorrow and then revert if we can't get it going.